distributed/shuffle/_core.py

```diff
@@ -111,6 +112,17 @@ async def send(
             run_id=self.run_id,
         )
 
+    async def send(
+        self, address: str, shards: list[tuple[_T_partition_id, bytes]]
+    ) -> None:
+        return await retry(
+            partial(self._send, address, shards),
+            count=3,
+            delay_min=1,
+            delay_max=5,
```

Comment:
Thinking about it, adding config options for these values might make testing easier/faster. 

I think we can also be more liberal with the retry count. If the shuffle fails or gets aborted, `raise_if_closed` in `_send` would interrupt the retry mechanism.